File: ts_sdk/src/resource_management/fs_records/claude/claude-settings/skill-usage.ts
/**
 * ClaudeSkillUsageFsRecord — skill usage tracking from claude settings.
 */
import { FsRecord, type FsRecordData } from '../../fs-record';
import { RecordType } from '../../record-types';
import { StorageLayout } from '../../storage-layout';
import { fsRecordTypeRegistry } from '../../record-type-registry';

export class ClaudeSkillUsageFsRecord extends FsRecord {
  static override _recordType = RecordType.CLAUDE_SETTINGS_SKILL_USAGE;
  static override _readOnly = true;
  static override _storageLayout = StorageLayout.LIST_ITEM;

  skill_name?: string;
  use_count = 0;
  last_used?: string;

  constructor(data?: Partial<FsRecordData>) {
    super(data);
  }
}

fsRecordTypeRegistry.register(ClaudeSkillUsageFsRecord._recordType, ClaudeSkillUsageFsRecord as any);

File: flow_sdk/fs_records/claude/claude_settings/skill_usage.py
"""ClaudeSkillUsageFsRecord -- per-skill usage entry from ~/.claude.json."""

from __future__ import annotations

from typing import Any, ClassVar

from flow_sdk.fs_store import Record, RecordType


class ClaudeSkillUsageFsRecord(Record):
    """Usage statistics for a single skill."""

    _record_type: ClassVar[str] = RecordType.CLAUDE_SETTINGS_SKILL_USAGE

    def __init__(self, **kwargs: Any):
        if "type" not in kwargs:
            kwargs["type"] = RecordType.CLAUDE_SETTINGS_SKILL_USAGE
        super().__init__(**kwargs)
        from flow_sdk.fs_store.fs_ref import FSRef
        object.__setattr__(self, "_asset_ref", FSRef("/", read_only=True))
        if self.skill_name and not object.__getattribute__(self, "__dict__").get("id"):
            object.__getattribute__(self, "__dict__")["id"] = self.skill_name

    @property
    def skill_name(self) -> str:
        return object.__getattribute__(self, "__dict__").get("skill_name") or ""

    @classmethod
    def from_raw(cls, skill_name: str, data: dict) -> ClaudeSkillUsageFsRecord:
        """Create from a skillUsage entry."""
        rec = cls(
            skill_name=skill_name,
            usage_count=data.get("usageCount", 0),
            last_used_at=data.get("lastUsedAt", 0),
        )
        rec.id = skill_name
        rec.name = skill_name
        return rec
